tests/imfile-file-descriptor-exhasution.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# This file is part of the rsyslog testbench, released  under ASL 2.0
```

Comment:
I may be wrong, but I think the test can actually never fail. Did you forget the check?

Reply:
My thought was, as this test is to open more files than the FD limit set for this process. 
So, if we cross that limit the process should not crash and should gracefully exit. So, in case the process crashed, then this test case will be marked failed.

I can add a test if the FD limit did not exhaust, but that won't be actually covering the scenario which we fixed.

Pls feel free to share your thoughts.

Reply:
> So, in case the process crashed, then this test case will be marked failed.

That's OK, but you do a content check for the error message in any case. So we should ensure that the message is present when we shut down rsyslog. Along the same lines, it probably makes sense to explicitly set the fd ulimit to 1024 so that we can test even if the OS by default provides more (or is this somehow a bad idea - not sure).

Reply:
Have added a check to make sure if the process crashed or still running, and depending on it test will Pass or Fail.

Also as you mentioned, i think its better to keep the fd limit 1024 for the specific PID when we are running the test. 

I executed the test with the failure and with changes and it looks fine.

Let me know your comments on the same.

Reply:
Also, as the crash was in process_symlink(), so i have modified the test case to exactly go through the same behaviour.